### PR TITLE
Make chat apis use postgres models

### PR DESCRIPTION
**Pull request by gcapalbo**
Hold off on merge for now, want to add a blocking migration before we merge this.

@benrudolph

### CODE REVIEW

File: corehq/apps/sms/views.py [hunk 1]
```diff
@@ -79,7 +82,7 @@
 from dimagi.utils.decorators.view import get_file
 from dimagi.utils.logging import notify_exception
 from dimagi.utils.web import json_response
-from dimagi.utils.couch import release_lock
+from dimagi.utils.couch import release_lock, CriticalSection
```
Comment (dimagimon) [inline corehq/apps/sms/views.py]: F401 'release_lock' imported but unused

File: corehq/apps/sms/models.py [hunk 2]
```diff
@@ -490,6 +490,42 @@ class Meta:
...
+    def by_user(cls, domain, user_id, contact_id):
+        """
+        Returns the SQLLastReadMessage representing the last chat message
+        that was read in the given domain by the given user_id for the given
+        contact_id.
+        """
+        try:
+            # It's not expected that this can raise MultipleObjectsReturned
+            # since we lock out creation of these records with a CriticalSection.
+            # So if that happens, let the exception raise.
```
Comment (benrudolph) [inline corehq/apps/sms/models.py]: :+1:

File: corehq/apps/sms/views.py [hunk 3]
```diff
@@ -787,129 +790,183 @@ def _fmt(d):
...
+
+    @property
+    @memoized
+    def start_date(self):
+        if not self.start_date_str:
+            return None
+
+        try:
+            return iso_string_to_datetime(self.start_date_str)
+        except:
```
Comment (benrudolph) [inline corehq/apps/sms/views.py]: nit: may want to make this just `ValueError`, but if it doesn't matter too much and only cares about succeeding then fine to leave

File: corehq/apps/sms/views.py [hunk 4]
```diff
@@ -787,129 +790,183 @@ def _fmt(d):
...
+                lrm = SQLLastReadMessage.by_user(self.domain, request.couch_user.get_id, self.contact_id)
+
+            if lrm:
+                lrm_timestamp = json_format_datetime(lrm.message_timestamp)
+        return HttpResponse(json.dumps({
+            'message_timestamp': lrm_timestamp,
+        }))
 
 
 class DomainSmsGatewayListView(CRUDPaginatedViewMixin, BaseMessagingSectionView):
```
Comment (benrudolph) [inline corehq/apps/sms/views.py]: very nice. this file reads like :book:

**Comment by gcapalbo** [comment]
@benrudolph updated

**Comment by gcapalbo** [comment]
(this is good to merge now)

**Comment by benrudolph** [comment]
awesome, will wait for tests to run

**Pull request merged**